src/background/utils/overwriteWitnetNodeConfiguration.ts

```diff
@@ -1,30 +1,60 @@
 import path from 'path'
 import fs from 'fs-extra'
 
+const nodeUrlUntilCharacter = (character: string) =>
+  new RegExp('node_url =([^;]*)' + character)
+
+const nodeUrlsRegex = /g\[([^;]*)\]|\"([^;]*)\"/
+
+export function hasOldWitnetIp({
+  oldWitnetNodes,
+  currentNodeConfiguration,
+}: {
+  oldWitnetNodes: Array<string>
+  currentNodeConfiguration: Array<string>
+}) {
+  return currentNodeConfiguration.some(ip => oldWitnetNodes.includes(ip))
+}
+
 // Replace witnet nodes urls in witnet configuration file
 export function overwriteWitnetNodeConfiguration({
   sheikahPath,
   witnetConfigFileName,
   publicNodeUrls,
+  oldWitnetNodes,
 }: {
   sheikahPath: string
   witnetConfigFileName: string
   publicNodeUrls: Array<string>
+  oldWitnetNodes: Array<string>
 }) {
   const replacement = `node_url = ${JSON.stringify(publicNodeUrls)}\n`
     .replace("'", '')
     .trim()
```

Comment:
No need to calculate this string if hasOldWitnetIp returns false, so let's move it inside the if.